src/WebJobs.Extensions.DurableTask/EntityScheduler/MessageSorter.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// provides message ordering and deduplication of request messages (operations or lock requests)
+    /// that are sent to entities, from other entities, or from orchestrations.
+    /// </summary>
+    [JsonObject(MemberSerialization.OptIn)]
+    internal class MessageSorter
+    {
+        // don't update the reorder window too often since the garbage collection incurs some overhead.
+        private static readonly TimeSpan MinIntervalBetweenCollections = TimeSpan.FromSeconds(10);
+
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public Dictionary<string, DateTime> Sent { get; set; }
+
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public Dictionary<string, ReceiveBuffer> Received { get; set; }
+
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public DateTime ReceiveHorizon { get; set; }
+
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public DateTime SendHorizon { get; set; }
+
+        public int NumberBufferedRequests =>
```

Comment:
This looks to be only used by our test code, which means you can make it internal. Also, adding a comment to indicate it is used for testing purposes would be nice.

Reply:
Sure.